.github/workflows/welcome-new-contributors.yaml

```diff
@@ -0,0 +1,25 @@
+name: 'Welcome New Contributors'
+
+on:
+  issues:
+    types: [opened]
+  pull_request_target:
+    types: [opened]
+
+jobs:
+  welcome-new-contributor:
+    runs-on: ubuntu-latest
+    steps:
+      - name: 'Greet the contributor'
+        uses: garg3133/welcome-new-contributors@v1.2
```

Comment:
Is this particular action needed? I ask because there seems to be similar action from "verified creator" here https://github.com/marketplace/actions/first-interaction, which might also be better maintained than https://github.com/garg3133/welcome-new-contributors which has not seen updates anymore. The "first-interaction" action seems to lack the `@contributor_name` macro though.

Reply:
That's a good question. I'll look into it. I've previously used the currently linked one in other repositories.

Reply:
I think adding a little personal touch of having the user's name is worth using the older action. It seems to stand alone well enough and appears to lack a need to be updated.

What are your thoughts?


An alternate solution is to fork that repository. Would that be preferable?

Reply:
I'm ok with this action, but since it is given `GITHUB_TOKEN` it would be bit safer if we could pin the hash. In that case the developer cannot publish a new version with the same tag, that does something bad.

Right now I don't find particularly good instructions for pinning action versions but here is one https://infra.apache.org/github-actions-policy.html.

Reply:
So in this case we would pin to this release?
https://github.com/garg3133/welcome-new-contributors/commit/a38583ed8282e23d63d7bf919ca2d9fb95300ca6

garg3133/welcome-new-contributors@a38583ed8282e23d63d7bf919ca2d9fb95300ca6

Reply:
Ahhh. Awesome. Looks like GitHub did its magic on mine and shortened it. I'll push a new commit.